### ISSUE TITLE
NullPointerException

### ISSUE THREAD

**Issue by pmakin** [North-West-Wind/CurseForge-CLI#19]
not duplicate of #18
After downloading [modpack](https://www.curseforge.com/minecraft/modpacks/minecolonies-official)  with id `250424` i get error: 

```
java.lang.NullPointerException: Cannot read the array length because "array" is null
	at java.base/java.util.Arrays.stream(Arrays.java:5428)
	at ml.northwestwind.Modpack.getModVersion(Modpack.java:202)
	at ml.northwestwind.Modpack.install(Modpack.java:95)
	at ml.northwestwind.Modpack.run(Modpack.java:29)
	at ml.northwestwind.Main.main(Main.java:28)
```

I even compiled binary from last commit (with god's help), but same error.

I don't know if i'm missing any files or settings, but this really worries me 😖 

Full log:
```
[paul@fedora curseforge-cli]$ ./curseforge modpack install 250424
CurseForge CLI v1.4.0
Installing MineColonies Official...
Starting mod download
[1/97] [S/F/S: 1/0/0] Downloaded [Forge 1.19.1/2] v6.2.43
[2/97] [S/F/S: 2/0/0] Downloaded TerraBlender Forge 1.19.2-2.0.1.125
[3/97] [S/F/S: 3/0/0] Downloaded Biomes O' Plenty 1.19.2-17.1.1.150
[4/97] [S/F/S: 4/0/0] Downloaded Ecologics 2.1.9 (Forge 1.19.1/1.19.2)
[5/97] [S/F/S: 5/0/0] Downloaded wwoo_forge-3.1.7+1.19.2.jar
[6/97] [S/F/S: 6/0/0] Downloaded DungeonsArise-1.19-2.1.51-beta.jar
[7/97] [S/F/S: 7/0/0] Downloaded journeymap-1.19.1-5.9.0beta2-forge
[8/97] [S/F/S: 8/0/0] Downloaded Neat-1.19-32.jar
[9/97] [S/F/S: 9/0/0] Downloaded comforts-forge-6.0.0+1.19.2.jar
[10/97] [S/F/S: 10/0/0] Downloaded kleeslabs-forge-1.19-12.1.0.jar
[11/97] [S/F/S: 11/0/0] Downloaded balm-4.5.3.jar
[12/97] [S/F/S: 12/0/0] Downloaded Vanilla+ Tools-1.19 1.1
[13/97] [S/F/S: 13/0/0] Downloaded shieldmechanics-1.19.2-3.5.jar
[14/97] [S/F/S: 14/0/0] Downloaded memorysettings-1.19.2-2.2.jar
[15/97] [S/F/S: 15/0/0] Downloaded screenscale-1.19.2-2.3.jar
[16/97] [S/F/S: 16/0/0] Downloaded fallingthrough-1.19.2-2.6.jar
[17/97] [S/F/S: 17/0/0] Downloaded hostilevillages-1.19.2-4.1.jar
[18/97] [S/F/S: 18/0/0] Downloaded lootintegrations-1.19.1-1.6.jar
[19/97] [S/F/S: 19/0/0] Downloaded smoothchunk-1.19.1-2.0.jar
[20/97] [S/F/S: 20/0/0] Downloaded limitedchunks-1.19.2-2.4.jar
[21/97] [S/F/S: 21/0/0] Downloaded brutalbosses-1.19.1-5.5.jar
[22/97] [S/F/S: 22/0/0] Downloaded betterfpsdist-1.19.1-1.6.jar
[23/97] [S/F/S: 23/0/0] Downloaded dusk-1.19.1-2.2.jar
[24/97] [S/F/S: 24/0/0] Downloaded dynview-1.19.1-2.9.jar
[25/97] [S/F/S: 25/0/0] Downloaded farsight-1.19.1-2.0.jar
[26/97] [S/F/S: 26/0/0] Downloaded connectivity-1.19.1-3.3.jar
[27/97] [S/F/S: 27/0/0] Downloaded logprot-1.19.1-1.8.jar
[28/97] [S/F/S: 28/0/0] Downloaded TreeChop-1.19.1-0.15.8.jar
[29/97] [S/F/S: 29/0/0] Downloaded EnchantmentDescriptions-Forge-1.19.2-13.0.1.jar
[30/97] [S/F/S: 30/0/0] Downloaded cookingforblockheads-forge-1.19-13.2.1.jar
[31/97] [S/F/S: 31/0/0] Downloaded cleancut-mc1.19-4.2-forge.jar
[32/97] [S/F/S: 32/0/0] Downloaded [1.19.2] Corpse 1.19.2-1.0.0
[33/97] [S/F/S: 33/0/0] Downloaded BHMenu-Forge-1.19.2-1.3.jar
[34/97] [S/F/S: 34/0/0] Downloaded AdditionalStructures v.4.0.1
[35/97] [S/F/S: 35/0/0] Downloaded Version 1.8.2 (1.19.2) (FORGE + FABRIC/QUILT)
[36/97] [S/F/S: 36/0/0] Downloaded Naturalist 2.1.1 (Forge 1.19-1.19.2)
[37/97] [S/F/S: 37/0/0] Downloaded Enhanced-Farming-1.19-1.4.5.jar
[38/97] [S/F/S: 38/0/0] Downloaded MVS 3.1.2 1.19 [FORGE]
[39/97] [S/F/S: 39/0/0] Downloaded PlayerRevive_FORGE_v2.0.9_mc1.19.2.jar
[40/97] [S/F/S: 40/0/0] Downloaded InfusedFoods - v7.1.4 for Minecraft 1.19.2
[41/97] [S/F/S: 41/0/0] Downloaded Not Enough Crashes 4.1.8+1.19.2 Forge
[42/97] [S/F/S: 42/0/0] Downloaded zmedievalmusic-1.19.2-1.4.jar
[43/97] [S/F/S: 43/0/0] Downloaded [1.19+] Mouse Tweaks 2.23 Forge
[44/97] [S/F/S: 44/0/0] Downloaded craftingtweaks-forge-1.19-15.1.0.jar
[45/97] [S/F/S: 45/0/0] Downloaded [Forge] Creeper Overhaul 2.0.4
[46/97] [S/F/S: 46/0/0] Downloaded [Forge 1.19-1.19.2] v8.0.75
[47/97] [S/F/S: 47/0/0] Downloaded Global Packs - Forge - 1.19.2 - v1.14.2
[48/97] [S/F/S: 48/0/0] Downloaded Hourglass v1.2.1.1 for 1.19.1 - 1.19.2
[49/97] [S/F/S: 49/0/0] Downloaded curios-forge-1.19.2-5.1.1.0.jar
[50/97] [S/F/S: 50/0/0] Downloaded Copper Drops 1.0 - 1.19+
[51/97] [S/F/S: 51/0/0] Downloaded cursery-1.19.2-3.0.jar
[52/97] [S/F/S: 52/0/0] Downloaded dragonfight-1.19.2-2.4.jar
[53/97] [S/F/S: 53/0/0] Downloaded CreativeCore_FORGE_v2.8.5_mc1.19.2.jar
[54/97] [S/F/S: 54/0/0] Downloaded ScalingHealth-1.19.2-7.0.0+6.jar
[55/97] [S/F/S: 55/0/0] Downloaded caveore-1.19.2-2.3.jar
[56/97] [S/F/S: 56/0/0] Downloaded Oh_The_Biomes_You'll_Go-forge-1.19.2-2.0.0.10.jar
[57/97] [S/F/S: 57/0/0] Downloaded trashslot-forge-1.19-12.0.0.jar
[58/97] [S/F/S: 58/0/0] Downloaded L_Ender's Cataclysm-0.20-1.19.2 hotfix
[59/97] [S/F/S: 59/0/0] Downloaded Corail Woodcutter 2.5.0 for 1.19.2
[60/97] [S/F/S: 60/0/0] Downloaded Jade-1.19.1-forge-8.1.1.jar
[61/97] [S/F/S: 61/0/0] Downloaded Library ferret 3.0.0 - Forge 1.19.2
[62/97] [S/F/S: 62/0/0] Downloaded sophisticatedbackpacks-1.19.2-3.18.24.703.jar
[63/97] [S/F/S: 63/0/0] Downloaded silent-lib-1.19.2-7.0.2.jar
[64/97] [S/F/S: 64/0/0] Downloaded sophisticatedcore-1.19.2-0.5.9.117.jar
[65/97] [S/F/S: 65/0/0] Downloaded Bookshelf-Forge-1.19.2-16.1.5.jar
[66/97] [S/F/S: 66/0/0] Downloaded The Graveyard 1.12 (FORGE) for 1.19.2
[67/97] [S/F/S: 67/0/0] Downloaded AmbientSounds_FORGE_v5.2.1_mc1.19.jar
[68/97] [S/F/S: 68/0/0] Downloaded [1.19.x]-Epic-Knights-Armor-and-Weapons-6.6v.jar
[69/97] [S/F/S: 69/0/0] Downloaded clickadv-1.19.2-2.9.jar
[70/97] [S/F/S: 70/0/0] Downloaded Copperized 2.5 - 1.19+
[71/97] [S/F/S: 71/0/0] Downloaded Goblin Traders 1.7.3
[72/97] [S/F/S: 72/0/0] Downloaded Storage Racks 1.19.2 1.1.4
[73/97] [S/F/S: 73/0/0] Downloaded Awesome dungeon edition ocean 3.0.0 - Forge 1.19.2
[74/97] [S/F/S: 74/0/0] Downloaded Awesome dungeon nether 3.0.0 - Forge 1.19.2
[75/97] [S/F/S: 75/0/0] Downloaded jei-1.19.2-forge-11.3.0.262.jar
[76/97] [S/F/S: 76/0/0] Downloaded geckolib-forge-1.19-3.1.19.jar
[77/97] [S/F/S: 77/0/0] Downloaded Awesome dungeon the end 3.0.0 - Forge 1.19.2
[78/97] [S/F/S: 78/0/0] Downloaded 1.5.2 - Forge(1.19 - fixed)
[79/97] [S/F/S: 79/0/0] Downloaded unstructured-0.5.2[FORGE].jar
[80/97] [S/F/S: 80/0/0] Downloaded DeathCounter-1.19.2-1.2.1.jar
[81/97] [S/F/S: 81/0/0] Downloaded alexsmobs-1.20.1
[82/97] [S/F/S: 82/0/0] Downloaded SoL-Carrot-1.19.2-1.14.0.jar
[83/97] [S/F/S: 83/0/0] Downloaded HarderSpawners-1.19.2-40.2.23.0.jar
[84/97] [S/F/S: 84/0/0] Downloaded citadel-1.12.11-1.19
[85/97] [S/F/S: 85/0/0] Downloaded spoornarmorattributes-2.0.6-1.19-forge
[86/97] [S/F/S: 86/0/0] Downloaded spoornweaponattributes-4.0.9-1.19-forge
[87/97] [S/F/S: 87/0/0] Downloaded blockui-1.19-0.0.64-ALPHA.jar
[88/97] [S/F/S: 88/0/0] Downloaded multi-piston-1.19.2-1.2.19-ALPHA.jar
[89/97] [S/F/S: 89/0/0] Downloaded polymorph-forge-0.45.0+1.19.2.jar
[90/97] [S/F/S: 90/0/0] Downloaded Farmer's Delight 1.2 - 1.19+
[91/97] [S/F/S: 91/0/0] Downloaded Delightful 3.1.1 - 1.19+
[92/97] [S/F/S: 92/0/0] Downloaded [Forge 1.19+] 1.1.3
[93/97] [S/F/S: 93/0/0] Downloaded ferritecore-5.0.2-forge.jar
[94/97] [S/F/S: 94/0/0] Downloaded minecolonies-1.19.2-1.0.1113-ALPHA.jar
[95/97] [S/F/S: 95/0/0] Downloaded structurize-1.19.2-1.0.460-ALPHA.jar
[96/97] [S/F/S: 96/0/0] Downloaded domum_ornamentum-1.19-1.0.75-ALPHA-universal.jar
[97/97] [S/F/S: 97/0/0] Downloaded appleskin-forge-mc1.19-2.4.2.jar
Iterated through 97 projects. 97 success, 0 failed, 0 skipped
java.lang.NullPointerException: Cannot read the array length because "array" is null
	at java.base/java.util.Arrays.stream(Arrays.java:5428)
	at ml.northwestwind.Modpack.getModVersion(Modpack.java:202)
	at ml.northwestwind.Modpack.install(Modpack.java:95)
	at ml.northwestwind.Modpack.run(Modpack.java:29)
	at ml.northwestwind.Main.main(Main.java:28)
```
Java:
```
openjdk version "17.0.4.1" 2022-08-12
OpenJDK Runtime Environment (Red_Hat-17.0.4.1.1-1.fc36) (build 17.0.4.1+1)
OpenJDK 64-Bit Server VM (Red_Hat-17.0.4.1.1-1.fc36) (build 17.0.4.1+1, mixed mode, sharing)
```

**Comment by North-West-Wind** [comment]
I think this is due to the `.minecraft/versions` directory missing as the program assumes the user to have run the Minecraft client for at least once. I should be able to fix that soon.

**Comment by North-West-Wind** [comment]
Fixed in v1.4.1

**Comment by pmakin** [comment]
> I think this is due to the `.minecraft/versions` directory missing as the program assumes the user to have run the Minecraft client for at least once. I should be able to fix that soon.

I'm using MultiMc, that's the reason probably
